tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/event/IdentityProvisionListener.java

```diff
@@ -0,0 +1,221 @@
+            .map(Throwing.function(Filter::create).sneakyThrow());
+        this.firstnameAttribute = listenerConfig.getString("firstnameAttribute", "givenName");
+        this.surnameAttribute = listenerConfig.getString("surnameAttribute", "sn");
+        this.usernameAttribute = ldapConfiguration.getUsernameAttribute().orElse(ldapConfiguration.getUserIdAttribute());
+    }
+
+    @VisibleForTesting
+    public IdentityProvisionListener(LdapRepositoryConfiguration ldapConfiguration,
+                                     IdentityRepository identityRepository) throws LDAPException {
+        this.ldapConnectionPool = new LDAPConnectionFactory(ldapConfiguration).getLdapConnectionPool();
+        this.ldapConfiguration = ldapConfiguration;
+        this.identityRepository = identityRepository;
+        this.objectClassFilter = Filter.createEqualityFilter("objectClass", ldapConfiguration.getUserObjectClass());
+        this.userExtraFilter = Optional.ofNullable(ldapConfiguration.getFilter())
+            .map(Throwing.function(Filter::create).sneakyThrow());
+        this.firstnameAttribute = "givenName";
+        this.surnameAttribute = "sn";
+        this.usernameAttribute = ldapConfiguration.getUsernameAttribute().orElse(ldapConfiguration.getUserIdAttribute());
+    }
+
+    @Override
+    public Group getDefaultGroup() {
+        return GROUP;
+    }
+
+    @Override
+    public boolean isHandling(Event event) {
+        return isInboxCreated(event);
+    }
+
+    private boolean isInboxCreated(Event event) {
+        if (event instanceof MailboxEvents.MailboxAdded mailboxAdded) {
+            return mailboxAdded.getMailboxPath().getName().equalsIgnoreCase(MailboxConstants.INBOX)
+                && mailboxAdded.getMailboxPath().getUser().equals(event.getUsername());
+        }
+        return false;
+    }
+
+    @Override
+    public Publisher<Void> reactiveEvent(Event event) {
+        if (!isHandling(event)) {
+            return Mono.empty();
+        }
+
+        Username username = event.getUsername();
+        return Mono.fromCallable(() -> searchLdap(event, username))
+            .map(Throwing.function(this::asIdentityCreationRequest))
+            .flatMap(identityCreationRequest -> hasNoDefaultIdentity(username)
+                .flatMap(noDefaultIdentity -> Mono.from(identityRepository.save(username, identityCreationRequest))
+                    .doOnSuccess(identity -> LOGGER.info("Created default identity for user {} with display name {}", username.asString(), identity.name()))))
+            .then()
+            .onErrorResume(e -> {
+                LOGGER.error("Unexpected error during JMAP identity provisioning for user {}", event.getUsername(), e);
+                return Mono.empty();
+            })
+            .subscribeOn(ReactorUtils.BLOCKING_CALL_WRAPPER);
+    }
+
+    private SearchResultEntry searchLdap(Event event, Username username) throws LDAPSearchException {
```

Comment:
Can drop event here? The only value you access from the event is username, that you pass as an argument already and use